als-actions/shared/src/main/scala/org/mulesoft/als/actions/hover/PatchedHover.scala

```diff
@@ -1,34 +1,42 @@
 package org.mulesoft.als.actions.hover
 
 import amf.aml.client.scala.model.document.Dialect
+import amf.aml.client.scala.model.domain.ClassTerm
 import amf.core.client.common.position.{Range => AmfRange}
 import amf.core.client.scala.model.domain.AmfObject
 import amf.core.client.scala.vocabulary.ValueType
+import amf.core.internal.annotations.Aliases
 import org.mulesoft.als.common.YPartBranch
 import org.mulesoft.amfintegration.AmfImplicits.AmfObjectImp
 import org.mulesoft.amfintegration.dialect.dialects.asyncapi20.AsyncApi20Dialect
 import org.mulesoft.amfintegration.dialect.dialects.oas.{OAS20Dialect, OAS30Dialect}
 import org.mulesoft.amfintegration.dialect.dialects.raml.raml08.Raml08TypesDialect
 import org.mulesoft.amfintegration.dialect.dialects.raml.raml10.Raml10TypesDialect
-import org.mulesoft.amfintegration.vocabularies.integration.VocabularyProvider
+import org.mulesoft.amfintegration.vocabularies.integration.{AlsVocabularyRegistry, VocabularyProvider}
+import org.mulesoft.amfintegration.vocabularies.propertyterms.patched.raml.raml10.Raml10UsesKeyTerm
 import org.yaml.model.YMapEntry
 
 class PatchedHover(provider: VocabularyProvider) {
 
-  def getHover(obj: AmfObject, branch: YPartBranch, dialect: Dialect): Option[(Seq[String], Option[AmfRange])] =
+  def getHover(obj: AmfObject,
+               branch: YPartBranch,
+               dialect: Dialect,
+               aliases: Option[Aliases]): Option[(Seq[String], Option[AmfRange])] =
     obj.metaURIs.headOption.flatMap(metaUri => {
       branch.parentEntry match {
-        case Some(entry: YMapEntry) => getPatchedHover(metaUri, entry, dialect.id)
+        case Some(entry: YMapEntry) => getPatchedHover(metaUri, entry, dialect.id, aliases)
         case _                      => None
       }
     })
 
   private def getPatchedHover(metaUri: String,
                               entry: YMapEntry,
-                              dialectId: String): Option[(Seq[String], Option[AmfRange])] = {
-    val dialectName = dialectNames.getOrElse(dialectId, "unknown")
-    val valueType   = buildTerm(ValueType(metaUri), entry.key.toString, dialectName)
-    provider
+                              dialectId: String,
+                              aliases: Option[Aliases]): Option[(Seq[String], Option[AmfRange])] = {
+    val dialectName   = dialectNames.getOrElse(dialectId, "unknown")
+    val valueType     = buildTerm(ValueType(metaUri), entry.key.toString, dialectName)
+    val hoverProvider = indexAliasTerms(valueType, aliases, dialectName, ValueType(metaUri).name)
```

Comment:
create a companion object to provide the `provider.branch` as parameter, making the class constructor private so it must be accessed via the companion

for example:
```
private val patches: Seq[() => Unit] = Seq(
  () => baseUnit.annotations.aliases.foreach(aliases => indexAliasTerms(valueType, aliases, dialectName, ValueType(metaUri).name)
)

def init(): Unit => patches.foreach(_())
```

then there should be a companion object which does something like
```
object PatchedHover{
  def apply(provider...) = {
    val ph = new PatchedHover(provider.branch)
    ph.init()
    ph
  }
}
```

This should make the next patch easier as one should only add the lambda to the patches and it should ensure it patches and initializes everywhere